File: src/MrHotel/ApiService/Reports/Data/StockAdjustmentData.cs
namespace MrHotel.ApiService.Reports.Data;

using MrHotel.Database.Entities.Reports;

public record StockAdjustmentData(
    Guid ProductId,
    int Quantity)
{
    public StockAdjustment ToStockAdjustment()
    {
        return new() { Quantity = this.Quantity, ProductId = this.ProductId };
    }
}

File: src/MrHotel/WebApp/src/app/services/report-manager/data/stock-adjustment-data.ts
import { Guid } from "@customTypes/.";

export class StockAdjustmentData {
  public constructor(
    private readonly _productId: Guid,
    private readonly _quantity: number,
  ) {}

  public get id(): Guid {
    return this._productId;
  }

  public get quantity(): number {
    return this._quantity;
  }

  public toJSON(): Record<string, string> {
    return {
      productId: this._productId,
      quantity: this._quantity.toString(),
    };
  }
}
